src/pybind/mgr/test_orchestrator/module.py

```diff
@@ -196,9 +197,9 @@ def get_inventory(self, node_filter=None):
             c_v_out = check_output(['ceph-volume', 'inventory', '--format', 'json'])
         except OSError:
             cmd = """
-            . /tmp/ceph-volume-virtualenv/bin/activate
+            . {}/ceph-volume-virtualenv/bin/activate
             ceph-volume inventory --format json
-            """
+            """.format(tmpdir=os.environ.get('TMPDIR', '/tmp'))
```

Comment:
here, the mgr really don't have any way to access `MGR_ORCHESTRATOR_CLI_VIRTUALENV` without very ugly hacks. do you have an idea?

Reply:
i am not sure how `MGR_ORCHESTRATOR_CLI_VIRTUALENV` is related to the path where `ceph-volume` is installed into.

probably you actually meant to say `${CEPH_VOLUME_VIRTUALENV}` ? in that case, i think we could set `TMPDIR` environmental variable when running `cmake` or add the path . or parse and export `CEPH_VOLUME_VIRTUALENV`, before executing `vstart.sh` in `run-backend-api-tests.sh`.

which way do you prefer?

Reply:
`run-backend-api-tests.sh` is just a helper script. Maybe adding this to `vstart.sh` might be less worse?

Reply:
why do you think `vstart.sh` should be aware of things like `CEPH_VOLUME_VIRTUALENV` ?

Reply:
´vstart.sh` already reads cmake variables, so this wouldn't be a completely new concept to `vstart`. And in a development environment, we somehow need make the mgr aware of the `CEPH_VOLUME_VIRTUALENV` path in order to find the c-v virtualenv.

Reply:
i don't think we are discussing how this setting should be read.  i am just trying to avoid dumping the changes like this to `vstart.sh`.

again, i'd suggest run test using `make` or any generator supported by cmake, or just use `ctest` . simpler this way.

Reply:
`ctest` doesn't execute this code path. Instead it's meant for general development use cases, where one wants to run a cluster that was deployed with `vstart. 

So yeah, moving the C-V virtualenv  will break this code :cry:

Reply:
> `ctest` doesn't execute this code path. 

well, it depends. see https://github.com/ceph/ceph/blob/a9dfd7067f8bee78bf225bcbe484470aaa1a044f/src/CMakeLists.txt#L460